native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/DefaultHttpResponseFuture.java

```diff
@@ -118,6 +120,11 @@ public void notifyHttpListener(HttpCarbonMessage httpCarbonMessage) {
     public void notifyHttpListener(Throwable throwable) {
         responseLock.lock();
         try {
+            if (backPressureHandler != null) {
+                backPressureHandler.getBackPressureObservable().removeListener();
+            } else if (outboundMsgHolder != null) {
+                outboundMsgHolder.getBackPressureObservable().removeListener();
+            }
```

Comment:
else we can print a warning saying no BackPressureObservable found.

Reply:
Also, lets add a comment explaining why we have two of them.

Reply:
Resolved in [`557b8e3 `](https://github.com/ballerina-platform/module-ballerina-http/pull/837/commits/557b8e394a65a39f247b739acf498a71fd2b462e)